x-pack/metricbeat/module/aws/utils_test.go

```diff
@@ -284,3 +329,56 @@ func TestFindTimestamp(t *testing.T) {
 		assert.Equal(t, c.expectedTimestamp, outputTimestamp)
 	}
 }
+
+func TestFindIdentifierFromARN(t *testing.T) {
+	cases := []struct {
+		resourceARN        string
+		expectedIdentifier string
+	}{
+		{
+			"arn:aws:rds:eu-west-1:123456789012:db:mysql-db",
+			"mysql-db",
+		},
+		{
+			"arn:aws:ec2:us-east-1:123456789012:instance/i-123",
+			"i-123",
+		},
```

Comment:
[In theory](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) it is also possible to have ARNs whose resource ID is not namespaced, like `arn:partition:service:region:account-id:resource`, but I couldn't find any example. Also some services like S3 accept paths, that can have `/` (`arn:aws:s3:::my_corporate_bucket/exampleobject.png`).

Reply:
Please correct me if I misunderstood you 😬 For ARNs whose resourceID is not namespaced, do you mean something like `arn:aws:sns:us-east-1:627959692251:notification-topic-1` for SNS? This should be taken care of by `arn.Parse` https://github.com/elastic/beats/pull/12480/files#diff-738fa152f33c614c76cc3fabf326cc8fR190

Good point about service names with `/` in there like S3.  I need to think of something else for this. This best way will be use ARN as the identifier, which is natural for AWS resources. Probably worth spending some time to figure out how to put ARN as identifier here: https://github.com/elastic/beats/blob/master/x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go#L199

Reply:
> For ARNs whose resourceID is not namespaced, do you mean something like `arn:aws:sns:us-east-1:627959692251:notification-topic-1` for SNS? This should be taken care of by `arn.Parse` https://github.com/elastic/beats/pull/12480/files#diff-738fa152f33c614c76cc3fabf326cc8fR190

Yeah, I meant something like this one for SNS. I'd add a test case for this, to cover the processing we do with `arn.Resource` after `arn.Parse`.

Reply:
Will do! sns test case will be added in next commit. 

@jsoriano Turned out it is invalid to use '/' and ':' in S3 bucket names. I know you can use '/' for objects' names that are stored in S3 but not S3 bucket name itself.
<img width="1076" alt="Screen Shot 2019-06-12 at 4 46 39 PM" src="https://user-images.githubusercontent.com/14081635/59391560-e667ff80-8d31-11e9-9410-101df7153c38.png">

<img width="1090" alt="Screen Shot 2019-06-12 at 4 46 31 PM" src="https://user-images.githubusercontent.com/14081635/59391566-ea941d00-8d31-11e9-99a2-4c735ea3fac3.png">
So maybe we are ok here? 😬

Reply:
I guess so :slightly_smiling_face: